src/Microsoft.AspNet.OData.Shared/EdmChangedObjectCollection.cs

```diff
@@ -57,5 +69,288 @@ private void Initialize(IEdmEntityType entityType)
+            {
+                object value;
+                changedObj.TryGetPropertyValue(key.Name, out value);
+
+                if (value != null)
+                {
+                    keyValues.Add(key.Name, value);
+                }
+            }
+
+            return keyValues;
+        }
+
+        private void PatchItem(EdmStructuredObject changedObj, EdmStructuredObject originalObj)
+        {
+            foreach (string propertyName in changedObj.GetChangedPropertyNames())
+            {
+                ApplyProperties(changedObj, originalObj, propertyName);
+            }
+
+            foreach (string propertyName in changedObj.GetUnchangedPropertyNames())
+            {
+                ApplyProperties(changedObj, originalObj, propertyName);
+            }
+        }
+
+        private void ApplyProperties(EdmStructuredObject changedObj, EdmStructuredObject originalObj, string propertyName)
+        {
+            object value;
+            if (changedObj.TryGetPropertyValue(propertyName, out value))
+            {
+                EdmChangedObjectCollection changedColl = value as EdmChangedObjectCollection;
+                if (changedColl != null)
+                {
+                    TypelessPatchMethodHandler patchHandler = PatchHandler.GetNestedPatchHandler(originalObj, propertyName);
+                    if (patchHandler != null)
+                    {
+                        changedColl.Patch(patchHandler);
+                    }
+                    else
+                    {
+                        object obj;
+                        originalObj.TryGetPropertyValue(propertyName, out obj);
+
+                        ICollection<EdmStructuredObject> edmColl = obj as ICollection<EdmStructuredObject>;
+
+                        changedColl.Patch(edmColl);
+                    }
+                }
+                else
+                {
+                    //call patchitem if its single structuredobj
+                    EdmStructuredObject structuredObj = value as EdmStructuredObject;
+
+                    if (structuredObj != null)
+                    {
+                        object obj;
+                        originalObj.TryGetPropertyValue(propertyName, out obj);
+
+                        EdmStructuredObject origStructuredObj = obj as EdmStructuredObject;
+
+                        if(origStructuredObj == null)
+                        {
+                            if(structuredObj is EdmComplexObject)
+                            {
+                                origStructuredObj = new EdmComplexObject(structuredObj.ActualEdmType as IEdmComplexType);
+                            }
+                            else
+                            {
+                                origStructuredObj = new EdmEntityObject(structuredObj.ActualEdmType as IEdmEntityType);
+                            }
+
+                            originalObj.TrySetPropertyValue(propertyName, origStructuredObj);
```

Comment:
>TrySetPropertyValue [](start = 40, length = 19)

What if this fails?

Reply:
Since we use trygetpropertyvalue before, its very less likely to fail, even though would you suggest to throw a generic exception if this returns false? Also I dont see we are dealing that in other places where we use trysetpropertyvalue.